icu4c/source/i18n/number_simple.cpp

```diff
@@ -0,0 +1,86 @@
+// © 2017 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+
+#include "unicode/numberformatter.h"
+#include "number_formatimpl.h"
+#include "number_utils.h"
+#include "number_patternmodifier.h"
+
+using namespace icu;
+using namespace icu::number;
+using namespace icu::number::impl;
+
+
+SimpleNumberFormatter SimpleNumberFormatter::forLocale(const icu::Locale &locale, UErrorCode &status) {
+    return SimpleNumberFormatter::forLocaleAndGroupingStrategy(locale, UNUM_GROUPING_AUTO, status);
+}
+
+SimpleNumberFormatter SimpleNumberFormatter::forLocaleAndGroupingStrategy(
+    const icu::Locale &locale, UNumberGroupingStrategy groupingStrategy, UErrorCode &status) {
+    SimpleNumberFormatter retval;
+    retval.fSymbols = { new DecimalFormatSymbols(locale, status), status };
+    retval.fMicros = new SimpleMicroProps();
+    if (retval.fMicros == nullptr) {
+        status = U_MEMORY_ALLOCATION_ERROR;
+        return retval;
+    }
```

Comment:
check for U_FAILURE / return before you potentially clobber a failure code

Reply:
Resolved, at least to the extent of being consistent with other functions